### ISSUE TITLE
Sync Notes to OneDrive, Google Drive, MEGA

### ISSUE THREAD

**Issue by alphanerdgen** [msoultanidis/quillnote#65]
Hope they're on your roadmap (along with end-to-end encryption #59).

**Comment by ltGuillaume** [comment]
I would very much like to _not_ see this implemented to be honest. Stuff like this would make Quillnote rather bloat-ish, plus implementing APIs like this might even introduce privacy risks.

If you really want to use those sync services, why not sync a (n encrypted #59) backup folder that Quillnote would keep up-to-date on the phone's storage using their respective sync apps?

**Comment by CypherpunkSamurai** [comment]
I actually like the idea. I would like it to be a plugin if possible. Not in the app itself. As that'd bloat the app

**Comment by ltGuillaume** [comment]
But all these services have their own sync apps, right?

**Comment by CypherpunkSamurai** [comment]
They do. But, It's hard to maintain a battery efficient sync routine. Also using rclone with termux just for notes is a mess.

Can we please have a plugin system? Atleast a few plugins like Nextcloud, WebDAV, Mega and Dropbox.

**Comment by ltGuillaume** [comment]
Yeah, I just found out that Nextcloud's app does not sport an _automatic_ bidirectional sync feature, only a manual one. So, it might be so for a few other services mentioned here, too?

**Comment by CypherpunkSamurai** [comment]
> Yeah, I just found out that Nextcloud's app does not sport an _automatic_ bidirectional sync feature, only a manual one. So, it might be so for a few other services mentioned here, too?

Yes. Most cloud drives have one directional sync. When creating cloud sync feature always make one directional sync the default. Each sync service should have a flag for two directional sync.

One directional sync checks (check if file exists and was modified in past) should be default. In case two directional sync is supported disable that.

**Comment by yant0** [comment]
Would like to have this. my addition maybe upload an encrypted note to uguu.se 👍

**Comment by ltGuillaume** [comment]
> 
> 
> Yeah, I just found out that Nextcloud's app does not sport an _automatic_ bidirectional sync feature, only a manual one. So, it might be so for a few other services mentioned here, too?

Might have been wrong about this? Seems like Nextcloud keeps syncing automatically once you've ticked `Sync` for a file/folder, until you delete the file (`local only`).

**Comment by urizev** [comment]
I suggested Etebase sync support.

https://github.com/msoultanidis/quillnote/issues/35

IMHO, it is a great privacy sync service.

**Comment by msoultanidis** [comment]
Next update will feature the ability to sync notes locally, as files. That means you will be able to sync with any provider that supports files, including Nextcloud, with their application.
